src/game/enemies/marun.c

```diff
@@ -0,0 +1,509 @@
+
+        screenX += (enemy->region[0] << 8);
+        screenY += (enemy->region[1] << 8);
+
+        if (sub_8052394(screenY, screenX, 1, 8, NULL, sub_805217C) > 0) {
+            enemy->qPos.y += enemy->unk18;
+            enemy->unk18 += 0x20;
+        }
+    } else {
+        enemy->unk18 = 0;
+        sub_805CD70(&enemy->qPos, &enemy->qUnk1C, enemy->region, &enemy->unk9);
+    }
+
+    if (enemy->speed > 0) {
+        enemy->speed -= 4;
+    } else {
+        enemy->speed = 0;
+    }
+}
+
+bool32 sub_8063C98(Marun *enemy)
+{
+    Sprite *s = &enemy->s;
+    s32 worldX;
+    s32 worldY;
+    u8 i;
+
+    worldX = I(enemy->qPos.x);
+    worldY = I(enemy->qPos.y);
+    worldX = (TO_WORLD_POS_RAW(worldX, enemy->region[0]));
+    worldY = (TO_WORLD_POS_RAW(worldY, enemy->region[1]));
+
+    for (i = 0;; i++) {
+        Player *p;
+        s32 dx, y;
+
+        if (i > 1)
+            break;
+        p = sub_805CD20(i);
+        if (!p)
+            break;
+
+        if (s->frameFlags & SPRITE_FLAG_MASK_X_FLIP) {
+            dx = I(p->qWorldX) - worldX - 1;
+            if ((u32)dx > 0x4E)
+                goto x_flip_cleanup;
+
+            y = I(p->qWorldY);
+            if (y < worldY + 0x50 && y > worldY - 0x10) {
+                return TRUE;
+            }
+
+        x_flip_cleanup:
+            if (s->frameFlags & SPRITE_FLAG_MASK_X_FLIP) {
+                continue;
+            }
+        }
+
+        dx = worldX - I(p->qWorldX) - 1;
+        if ((u32)dx > 0x4E)
+            continue;
+
+        if (I(p->qWorldY) >= worldY + 0x50)
+            continue;
+        if (I(p->qWorldY) <= worldY - 0x10)
+            continue;
+
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+bool32 sub_8063D38(Marun *enemy)
+{
+    EnemyUnknownStruc0 unk;
+    bool32 result;
+
+    if (enemy->unk8 != TRUE) {
+        enemy->s.hitboxes[0].b.left -= 0x10;
+        enemy->s.hitboxes[0].b.top -= 0x10;
+        enemy->s.hitboxes[0].b.right -= 0x10;
+        enemy->s.hitboxes[0].b.bottom -= 0x10;
+
+        enemy->reserved.b.left -= 0x10;
+        enemy->reserved.b.top -= 0x10;
+        enemy->reserved.b.right -= 0x10;
+        enemy->reserved.b.bottom -= 0x10;
+    }
+
+    if (enemy->unk8 == TRUE) {
+        enemy->s.hitboxes[0].b.left = 0xFC;
+        enemy->s.hitboxes[0].b.top = 0xFC;
+        enemy->s.hitboxes[0].b.right = 4;
+        enemy->s.hitboxes[0].b.bottom = 4;
+    }
+
+    unk.unk4 = sub_8063EDC(enemy, &unk);
+    unk.spr = &enemy->s;
+    unk.posX = enemy->qUnk1C.x;
+    unk.posY = enemy->qUnk1C.y;
+    unk.regionX = enemy->region[0];
+    unk.regionY = enemy->region[1];
+    unk.me = enemy->me;
+    *(u8 *)((char *)&unk + 8) = enemy->spriteX;
+
+    result = sub_805C280(&unk);
+
+    if (enemy->unk8 != TRUE) {
+        enemy->s.hitboxes[0].b.left += 0x10;
+        enemy->s.hitboxes[0].b.top += 0x10;
+        enemy->s.hitboxes[0].b.right += 0x10;
+        enemy->s.hitboxes[0].b.bottom += 0x10;
+
+        enemy->reserved.b.left += 0x10;
```

Comment:
Should be `hitboxes[1]` not `reserved`

Reply:
Sure. I updated it in other parts too, since it makes sense to be `hitboxes[1]`